src/components/visx/TimeSeriesChart.tsx

```diff
@@ -137,18 +137,19 @@ export const TimeSeriesChart = <Datum extends {}>({
   const chartRef = useRef<HTMLDivElement>(null);
 
   // Chart data
-  const { xAccessor, yAccessor } = series[0];
+  const { xAccessor, yAccessor } = series[0]!;
 
-  const earliestDatum = data?.[0];
-  const latestDatum = data?.[data.length - 1];
+  const atLeastOnePoint = data.length > 0;
+  const earliestDatum = data[0];
+  const latestDatum = data[data.length - 1];
 
   // Chart state
   const getClampedZoomDomain = useCallback(
     (unclamped: number) => {
       return clamp(
         Math.max(1e-320, Math.min(Number.MAX_SAFE_INTEGER, unclamped)),
         minZoomDomain,
-        xAccessor(latestDatum) - xAccessor(earliestDatum)
+        atLeastOnePoint ? xAccessor(latestDatum!) - xAccessor(earliestDatum!) : 0
```

Comment:
does this need to be added to dep array?

Reply:
It's missing a bunch of stuff downstream of data but depends on data so it should be fine I think.